### PR TITLE
Feat/#gh861 replace prism bootstrapper

### PR DESCRIPTION
**Pull request by SimonWoodRhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/COMET-IME-Community-Edition/pulls) open
- [x] I have verified that I am following the COMET-IME [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/COMET-IME-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description
Replaced prism bootstrapper.

Also had to replace mef service locator which was part of prism.

All prism nugets and references have been removed.

### CODE REVIEW

File: CDP4Composition/Composition/CDP4Bootstrapper.cs [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+        /// Represents the state of the <see cref="CDP4Bootstrapper"/>
+        /// </summary>
+        protected string status;
+
+        /// <summary>
+        /// Runs the bootstrapper
+        /// </summary>
+        public void Run()
+        {
+            UpdateBootstrapperStatus("Configuring catalogs");
```
Comment (alexatrhea) [inline CDP4Composition/Composition/CDP4Bootstrapper.cs:63]: please use explicit `this.` qualifier here and below and in the other methods of this file.

File: CDP4Composition/Composition/MefServiceLocator.cs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        }
+
+        /// <summary>
+        /// Resolves the instance of the requested service.
+        /// </summary>
+        /// <param name="serviceType">Type of instance requested.</param>
+        /// <returns>The requested service instance.</returns>
+        protected override object DoGetInstance(Type serviceType, string key)
+        {
+            IEnumerable<Lazy<object, object>> exports = this.compositionContainer.GetExports(serviceType, null, key);
```
Comment (alexatrhea) [inline CDP4Composition/Composition/MefServiceLocator.cs:62]: can be implicit `var`?

File: CDP4Composition/Composition/MefServiceLocator.cs [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+        /// <summary>
+        /// Resolves all the instances of the requested service.
+        /// </summary>
+        /// <param name="serviceType">Type of service requested.</param>
+        /// <param name="key">Name of registered service you want. May be null.</param>
+        /// <returns>Sequence of service instance objects.</returns>
+        protected override IEnumerable<object> DoGetAllInstances(Type serviceType)
+        {
+            List<object> instances = new List<object>();
```
Comment (alexatrhea) [inline CDP4Composition/Composition/MefServiceLocator.cs:82]: can be implicit `var`?

File: CDP4Composition/Composition/MefServiceLocator.cs [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+            IEnumerable<Lazy<object, object>> exports = this.compositionContainer.GetExports(serviceType, null, null);
```
Comment (alexatrhea) [inline CDP4Composition/Composition/MefServiceLocator.cs:84]: can be implicit `var`?

File: CDP4IME/Shell.xaml.cs [hunk 5]
```diff
@@ -53,10 +53,18 @@ public partial class Shell : DXRibbonWindow
...
-        public Shell()
+        /// </summary>
+        /// <param name="viewModel">The <see cref="ShellViewModel"/> for this view</param>
+        /// <param name="ribbonContentBuilder">The <see cref="IRibbonContentBuilder"/></param>
+        [ImportingConstructor]
+        public Shell(ShellViewModel viewModel, IRibbonContentBuilder ribbonContentBuilder)
         {
             this.InitializeComponent();
+
+            DataContext = viewModel;
```
Comment (alexatrhea) [inline CDP4IME/Shell.xaml.cs:64]: use `this.` qualifier

**Review by alexatrhea — CHANGES_REQUESTED** [review]
Minor code consistency changes.

**Review by alexatrhea — APPROVED** [review]

**Review by samatrhea — APPROVED** [review]
sooooooooo happy that PRISM is finally gone!

**Comment by samatrhea** [comment]
one small thing to consider, i propose we rename CDP4Bootstrapper to COMETBootstrapper

**Pull request merged**